.github/workflows/ci.yml

```diff
@@ -16,64 +24,66 @@ jobs:
       BUILD_TYPE: ${{ matrix.build-type }}
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
 
-    - name: 📁 Create build folder
-      run: cmake -E make_directory ${{runner.workspace}}/build
+      - name: 📚 Install necessary packages
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt update
+          sudo apt install -y cmake
```

Comment:
why did you add an installation for cmake? I don't believe it's necessary

Reply:
Apparently, https://github.com/nektos/act is not configured to use github's runners by default, and I was using it to test CI locally, which was failing due to missing dependenecies. And apparently, github images ship with a ginormous database of preinstalled packages (13G). Therefore, you are correct, it is not necessary. However, I suggest to keep this in order to not have to have the github's giant image with all the packages stored and updated locally on disk, because local CI runs become a pain.

Reply:
hmm interesting consideration. I tend to be minimalist, i.e. to only install what I really need, but this is fine